src/nvim/ui_compositor.c

```diff
@@ -706,15 +706,19 @@ static void ui_comp_grid_resize(UI *ui, Integer grid, Integer width, Integer hei
 
 static void ui_comp_event(UI *ui, char *name, Array args)
 {
-  Error err = ERROR_INIT;
   UIEventCallback *event_cb;
   bool handled = false;
-
+  const Error initerr = ERROR_INIT;
   map_foreach_value(&ui_event_cbs, event_cb, {
+    Error err = initerr;
```

Comment:
you could write `(Error)ERROR_INIT` here and get rid of the extra definition

Reply:
That doesn't compile.

Reply:
I tried the below:
```C
  map_foreach_value(&ui_event_cbs, event_cb, {
    Error err = (Error)ERROR_INIT;
    Object res = nlua_call_ref(event_cb->cb, name, args, false, &err);
    if (res.type == kObjectTypeBoolean && res.data.boolean == true) {
      handled = true;
    }
    if (ERROR_SET(&err)) {
      ELOG("Error while executing ui_comp_event callback: %s", err.msg);
    }
    api_clear_error(&err);
  })
```
but the compiler throws the error of : `Too many arguments provided to function-like macro invocation`. I believe it's because the preprocessor sees any comma as an arg separator to a macro, and the `ERROR_INIT` macro [has a comma in its value](https://github.com/andrewwillette/neovim-willette/blob/0ab8fa1371b59ef8d864b6ce8661fbfc4bb04fe7/src/nvim/api/private/defs.h#L15).

Reply:
This works in multiple already existing code paths in our code base. not sure what causes the preprocessor to throw a childish tantrum in this particular case.

Reply:
I don't see any examples of `map_foreach_value` with a macro with of value `{...,...}` in the repository.

There is [an example](https://github.com/andrewwillette/neovim-willette/blob/0ab8fa1371b59ef8d864b6ce8661fbfc4bb04fe7/src/nvim/channel.c#L940) where the `ADD` macro([link](https://github.com/andrewwillette/neovim-willette/blob/0ab8fa1371b59ef8d864b6ce8661fbfc4bb04fe7/src/nvim/api/private/helpers.h#L77-L78)) is called in the `map_foreach_value`, but I believe this is still valid because the comma exists inside parenthesis. The preprocessor understands values inside parens are separate function args.

The `ERROR_INIT` macro has a comma [inside brackets](https://github.com/andrewwillette/neovim-willette/blob/0ab8fa1371b59ef8d864b6ce8661fbfc4bb04fe7/src/nvim/api/private/defs.h#L15).

I could be wrong, but it is my understanding of the issue.

Reply:
So I found a solution in line with surrounding the macro in parens! I saw that `STRING_OBJECT` ([link](https://github.com/andrewwillette/neovim-willette/blob/a17b0c947ad128edb9d9c5c54a610107a71e7f0a/src/nvim/api/private/helpers.h#L33-L35)) was used inside other macros, for example `PUT` ([link](https://github.com/andrewwillette/neovim-willette/blob/a17b0c947ad128edb9d9c5c54a610107a71e7f0a/src/nvim/api/autocmd.c#L298)). I followed `STRING_OBJECT`'s pattern (of what looks to be like an anonymous function in C?) to get it to work.

@bfredl / @zeertzjq please let me know if you think the prior implementation is preferable. Thank you for your time.  I really want to get this merged.

Reply:
yes, that is the preferred pattern.

Reply:
@bfredl sorry, I'm not sure I understand. Are you saying I should revert the `ERROR_INIT` [parens update](https://github.com/neovim/neovim/pull/21147/commits/a17b0c947ad128edb9d9c5c54a610107a71e7f0a#diff-5b8e70a02d059fb67689dcdf1385f39c7a6916c2e2e486459301ec9c65da041fR15) or not? If not, can you retrigger the CI pipeline?